src/charm.py

```diff
@@ -64,6 +67,52 @@ def evaluate(self, csr: bytes, relation_id: int, requirer_csrs: list[RequirerCSR
         return True
 
 
+class LimitToFirstRequester:
+    """Filter the CSR as to only allow the first requester to get a specific identifier."""
+
+    DENY_MSG = "CSR denied for relation ID %d, %s '%s' already requested."
+
+    def __init__(self, reserved_identifiers: Mapping):
+        self._reserved_identifiers = reserved_identifiers
+
+    def evaluate(self, csr: bytes, relation_id: int, requirer_csrs: list[RequirerCSR]) -> bool:
+        """Accept the CSR if no other relation previously requested any covered identifiers.
+
+        Identifiers that need to be unique are the Subject, all Subject Alternative Names,
+        all Subject Alternative IPs and all Subject Alternative OIDs.
+        """
+        csr_object = x509.load_pem_x509_csr(csr)
+        subjects = [
```

Comment:
I've researched this a little bit, the subject of an x.509 certificate is of type x.501 type Name. This is defined in ISO/IEC 9594-2:2005. From that PDF:
"""
9.2 Names in general
A (directory) name is a construct that identifies a particular object from among the set of all objects. A name shall be
unambiguous, that is, denotes just one object.
...
Syntactically, each name for an object or entry is an ordered sequence of relative distinguished names (see 9.3)
"""

And 

"""
RelativeDistinguishedName ::= SET SIZE (1..MAX) OF AttributeTypeAndDistinguishedValue
AttributeTypeAndDistinguishedValue ::= SEQUENCE {
 type ATTRIBUTE.&id ({SupportedAttributes}),
 value ATTRIBUTE.&Type({SupportedAttributes}{@type}),
 primaryDistinguished BOOLEAN DEFAULT TRUE,
 valuesWithContext SET SIZE (1..MAX) OF SEQUENCE {
 distingAttrValue [0] ATTRIBUTE.&Type ({SupportedAttributes}{@type}) OPTIONAL,
 contextList SET SIZE (1..MAX) OF Context } OPTIONAL }
The set that forms an RDN contains exactly one AttributeTypeAndDistinguishedValue for each attribute which
contains distinguished values in the entry; that is, a given attribute type cannot appear twice in the same RDN.
"""

Reply:
All this to say I think there's only one CN in every certificate.